plan/plan.go

```diff
@@ -0,0 +1,75 @@
+package plan
+
+// DNSRecord holds information about a DNS record.
+type DNSRecord struct {
+	// The hostname of the DNS record
+	DNSName string
+	// The target the DNS record points to
+	Target string
+}
+
+// Plan can convert a list of desired and current records to a series of create,
+// update and delete actions.
+type Plan struct {
+	// List of current records
+	Current []DNSRecord
+	// List of desired records
+	Desired []DNSRecord
+
+	// The following lists hold actions to take in orer to move current state to
+	// desired state.
+
+	// Records that need to be created
+	Create []DNSRecord
+	// Records that need to be updated (current data)
+	UpdateOld []DNSRecord
+	// Records that need to be updated (desired data)
+	UpdateNew []DNSRecord
+	// Records that need to be deleted
+	Delete []DNSRecord
+}
+
+// Calculate computes the actions needed to move current state to desired state.
+func (p *Plan) Calculate() *Plan {
```

Comment:
maybe it makes to create a separate struct for records to be updated `ChangeList` or `RecordChangeList`, with the `create,update,delete` lists?